cparser.nim

```diff
@@ -506,6 +514,53 @@ proc declKeyword(p: Parser, s: string): bool =
     result = p.options.flags.contains(pfCpp)
   else: discard
 
+proc isAttribute(t: ref Token): tuple[isattr: bool, parens: uint] =
+  if t.xkind != pxSymbol:
+    return (isattr: false, parens: 0'u)
+  case t.s
+  of "__attribute__":
+    result = (isattr: true, parens: 2'u)
```

Comment:
would be nice if this could catch specific attributes like "deprecated"

Reply:
Is it a keyboard attribute? Many just expand to __attribute__ macros under the hood. If so that’d be preferred way to handle it imho. But if you can link a reference I could add it.

Reply:
https://renenyffenegger.ch/notes/development/languages/C-C-plus-plus/GCC/__attribute__/deprecated/index

Reply:
there's also a message form: https://gcc.gnu.org/onlinedocs/gcc/Common-Function-Attributes.html - we have the same in nim, so it could even auto-translate.

This is used in the llvm-c interface for example that nlvm uses - right now, it requires manual patching.

Reply:
Gotcha, yah I left that as a todo. It parses it, but would need plumbing to add it to pragmas, etc. Currently beyond my motivation for this PR. :) Though that and aligns would be pretty handy to add.